csrc/dynamic_type.h

```diff
@@ -214,6 +215,12 @@ struct DynamicType {
           return result;
         }(value)) {}
 
+  const std::type_info& type() const {
+    return std::visit(
```

Comment:
C++ question: why this can't be just:

```
return typeid(value_);
```

Reply:
`return typeid(value_);` would return the `type_info` for the `std::variant<...>`, but I actually wanted the `type_info` for the actual dtype. For example, if the variant holds a `double`, then it should return the `type_info` of `double`.

Reply:
Comment example added. Example error message:
```C++
using IntOrVec = DynamicType<Containers<std::vector>, int>;
IntOrVec v = 3;
(std::vector<double>)v;
// Cannot cast from i to St6vectorIdSaIdEE : incompatible type
```